### PR TITLE
feat: ITS v1

### PR DESCRIPTION
**Pull request by alanrsoares**

### CODE REVIEW

File: apps/maestro/src/features/AddErc20/hooks/useDeployAndRegisterRemoteStandardizedTokenMutation.ts [hunk 1]
```diff
@@ -102,14 +98,15 @@ export function useDeployAndRegisterRemoteStandardizedTokenMutation(
...
-        operator: deployer,
-        destinationChain: chainId,
-        gasValue: input.gasFees[i] ?? BigInt(0),
-      })
+    const registerTxData = input.destinationChainIds.map(
+      (destinationChain, i) =>
+        INTERCHAIN_TOKEN_FACTORY_ENCODERS.deployRemoteInterchainToken.data({
+          destinationChain: destinationChain,
+          gasValue: input.gasFees[i] ?? BigInt(0),
+          distributor: deployer,
```
Comment (alanrsoares) [inline apps/maestro/src/features/AddErc20/hooks/useDeployAndRegisterRemoteStandardizedTokenMutation.ts:106]: we should propmpt the user whether to allow the remote tokens to be mintable by the an address (default to deployer)

File: apps/maestro/src/features/ManageInterchainToken/actions/transfer-ownership/TransferInterchainTokenOwnership.tsx [hunk 2]
```diff
@@ -27,12 +27,21 @@ export const TransferInterchainTokenOwnership: FC = () => {
...
+    mode: "onChange",
+    reValidateMode: "onChange",
+  });
+
   const {
     writeAsync: transferOwnershipAsync,
     isLoading: isTransfering,
     data: transferResult,
-  } = useIerc20BurnableMintableTransferOwnership({
+  } = useInterchainTokenServiceTransferOperatorship({
```
Comment (milapsheth) [inline apps/maestro/src/features/ManageInterchainToken/actions/transfer-ownership/TransferInterchainTokenOwnership.tsx:42]: `transferDistributorship`
https://github.com/axelarnetwork/interchain-token-service/blob/e49d88463d42ec760176891df49fd31ade009360/contracts/interfaces/IDistributable.sol#L13

File: apps/maestro/src/features/RegisterRemoteTokens/hooks/useRegisterRemoteCanonicalTokens.ts [hunk 3]
```diff
@@ -56,13 +56,16 @@ export default function useRegisterRemoteCanonicalTokens(
...
       const gasValue = gasFees[i];
 
-      return encodeInterchainTokenServiceDeployRemoteCanonicalTokenData({
-        tokenId: tokenDetails.tokenId,
-        destinationChain: axelarChainId,
-        gasValue,
-      });
+      return INTERCHAIN_TOKEN_FACTORY_ENCODERS.deployRemoteCanonicalInterchainToken.data(
+        {
+          originalChain: sourceChain?.id ?? "0x",
```
Comment (milapsheth) [inline apps/maestro/src/features/RegisterRemoteTokens/hooks/useRegisterRemoteCanonicalTokens.ts:61]: we now allow deploying remote tokens from non origin chains as well. the token just needs to have been deployed on that chain first

File: apps/maestro/src/features/SendInterchainToken/hooks/useTokenManagerSendTokenMutation.ts [hunk 4]
```diff
@@ -46,18 +46,18 @@ export function useTokenManagerSendTokenMutation(
...
   const { data: tokenManagerAddress } =
-    useInterchainTokenServiceGetTokenManagerAddress({
+    useInterchainTokenServiceTokenManagerAddress({
       args: [config.tokenId],
       enabled: Boolean(config.tokenId),
     });
 
-  const { writeAsync: approveERC20Async, data: approveERC20Data } =
-    useIerc20BurnableMintableApprove({
+  const { writeAsync: approveInterchainTokenAsync, data: approveERC20Data } =
```
Comment (alanrsoares) [inline apps/maestro/src/features/SendInterchainToken/hooks/useTokenManagerSendTokenMutation.ts:54]: approve only needs to be called for canonical tokens

File: apps/maestro/src/features/RegisterRemoteTokens/hooks/useRegisterRemoteStandardizedTokens.ts
Comment (milapsheth) [inline apps/maestro/src/features/RegisterRemoteTokens/hooks/useRegisterRemoteStandardizedTokens.ts:1]: we can drop standardized naming everywhere now

Tokens can be of 3 types, `Interchain` (deployed by the service), `Canonical` (pre-existing), `Custom` (custom interchain tokens)

File: apps/maestro/src/features/RegisterRemoteTokens/hooks/useRegisterRemoteStandardizedTokens.ts [hunk 5]
```diff
@@ -46,26 +46,24 @@ export default function useRegisterRemoteStandardizedTokens(
...
     return destinationChainIds.map((chainId, i) =>
-      encodeInterchainTokenServiceDeployAndRegisterRemoteStandardizedTokenData({
+      // encode InterchainTokenService DeployAndRegisterRemoteStandardizedTokenData
+      INTERCHAIN_TOKEN_FACTORY_ENCODERS.deployRemoteInterchainToken.data({
         salt: tokenDeployment.salt,
-        name: tokenDeployment.tokenName,
-        symbol: tokenDeployment.tokenSymbol,
-        decimals: tokenDeployment.tokenDecimals,
+        originalChainName: sourceChain?.name ?? "",
         distributor: "0x", // remote tokens cannot be minted, so the distributor must be 0x
```
Comment (milapsheth) [inline apps/maestro/src/features/RegisterRemoteTokens/hooks/useRegisterRemoteStandardizedTokens.ts:61]: if distributor is not selected as `0` by user, we should check if it matches the distributor set on token on the current chain via `isDistributor`. For security, the factory doesn't allow changing the distributor to any address for future deployments.

File: apps/maestro/src/features/AddErc20/hooks/useDeployAndRegisterRemoteStandardizedTokenMutation.ts [hunk 6]
```diff
@@ -102,14 +98,15 @@ export function useDeployAndRegisterRemoteStandardizedTokenMutation(
...
+          originalChainName: input.sourceChainId,
+          salt: config.salt,
```
Comment (milapsheth) [inline apps/maestro/src/features/AddErc20/hooks/useDeployAndRegisterRemoteStandardizedTokenMutation.ts:108]: we also support minting to remote chains as part of the same multicall. See the integration test for examples. For canonical/custom tokens, an ERC20 approval to the factory is needed first.

**Pull request merged**